services/oauth/v2/oauth.go

```diff
@@ -27,38 +27,26 @@ func Init() {
 	loggerNm = "OAuthHandler"
 }
 
-func WithCache(cache Cache) func(*OAuthHandler) {
-	return func(h *OAuthHandler) {
-		h.cache = cache
-	}
-}
-
-func WithLocker(lock *rudderSync.PartitionRWLocker) func(*OAuthHandler) {
-	return func(h *OAuthHandler) {
-		h.lock = lock
-	}
-}
-
 // This function creates a new OauthErrorResponseHandler
-func NewOAuthHandler(provider tokenProvider, options ...func(*OAuthHandler)) *OAuthHandler {
+func NewOAuthHandler(provider TokenProvider, options ...func(*OAuthHandler)) *OAuthHandler {
 	cpTimeoutDuration := config.GetDuration("HttpClient.oauth.timeout", 30, time.Second)
 	oAuthHandler := &OAuthHandler{
-		tokenProvider: provider,
-		logger:        pkgLogger,
+		TokenProvider: provider,
+		Logger:        pkgLogger,
 		CpConn:        NewControlPlaneConnector(WithCpClientTimeout(cpTimeoutDuration), WithParentLogger(pkgLogger)),
 		// This timeout is kind of modifiable & it seemed like 10 mins for this is too much!
-		rudderFlowType:            RudderFlow_Delivery,
-		authStatusUpdateActiveMap: make(map[string]bool),
+		RudderFlowType:            RudderFlow_Delivery,
+		AuthStatusUpdateActiveMap: make(map[string]bool),
 	}
 	for _, opt := range options {
 		opt(oAuthHandler)
 	}
-	if oAuthHandler.cache == nil {
+	if oAuthHandler.Cache == nil {
 		cache := NewCache()
-		oAuthHandler.cache = cache
+		oAuthHandler.Cache = cache
 	}
-	if oAuthHandler.lock == nil {
-		oAuthHandler.lock = rudderSync.NewPartitionRWLocker()
+	if oAuthHandler.Lock == nil {
```

Comment:
The name `Lock` is a bit confusing as it coincides with `Lock()` method.
We need to update the name of property `Lock` to something like `Locker`, `Syncer` etc,. Let's do it that way ?